src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -393,19 +393,20 @@ export class SignAccountOpController extends EventEmitter {
     }
 
     // if the gasFeePayment is gas tank but the user doesn't have funds, disable it
-    const hasBalance = Object.keys(currentPortfolio).find((networkName) => {
-      if (networkName === 'gasTank' || networkName === 'rewards') return false
-
+    let balance = 0
+    Object.keys(currentPortfolio).forEach((networkName) => {
       const networkPortfolio = currentPortfolio[networkName]
-      if (!networkPortfolio) return false
-      if (!networkPortfolio.result) return false
-      if (!networkPortfolio.result.total) return false
-      if (!networkPortfolio.result.total.usd) return false
+      if (!networkPortfolio) return
+      if (!networkPortfolio.result) return
+      if (!networkPortfolio.result.total) return
+      if (!networkPortfolio.result.total.usd) return
```

Comment:
Minor, this could be shrinked to 1 condition with ts optional chaining `!networkPortfolio?.result?.total?.usd`. Also in case the total is `0`, we may (or may not) want to be truethful in this case.